### PR TITLE
Crash dumper

### PR DESCRIPTION
**Pull request by vext01**
These PR adds a crash dumper, and fixes a couple of other small bits. See individual commits.

Here's an example crash dump:
```
The trace compiler crashed!

Reason: Ran out of registers.

Executable code buffer:
  movabs rcx, 0xd
  mov rax, rcx
  movabs r8, 0
  mov r9, rcx

Register allocation (local -> reg):
  51 ->   8 (r8)
   7 ->   9 (r9)
   1 ->   1 (rcx)

thread 'tests::test_simple' panicked at 'stopped due to trace compilation error', ykcompile/src/lib.rs:222:9
```

### CODE REVIEW

File: ykcompile/src/lib.rs [hunk 1]
```diff
@@ -137,19 +229,31 @@ impl TraceCompiler {
...
-            let t = tt.op(i);
-            match t {
+            let res = match tt.op(i) {
                 TirOp::Statement(st) => tc.statement(st),
-                TirOp::Guard(_) => tc.nop(), // FIXME Implement guards.
+                TirOp::Guard(g) => tc.guard(g),
+            };
+
+            if let Err(e) = res {
+                err = Some(e);
```
Comment (ptersilie) [inline ykcompile/src/lib.rs]: Why don't we call `tc.crash_dump()` here so we don't need to assign the `err` variable.

**Comment by ptersilie** [comment]
Just one comment. Rest looks good.

File: ykcompile/src/lib.rs (hunk 1 above)
Comment (vext01) [reply ykcompile/src/lib.rs]: Fixed.

**Comment by vext01** [comment]
Ready for re-review.

**Comment by ptersilie** [comment]
Looks good. Please squash.

**Comment by vext01** [comment]
splat.

**Pull request merged**